wrappers/mongo/common_utils.go

```diff
@@ -0,0 +1,134 @@
+package epsagonmongo
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/epsagon/epsagon-go/epsagon"
+	"github.com/epsagon/epsagon-go/protocol"
+	"github.com/epsagon/epsagon-go/tracer"
+	"github.com/google/uuid"
+	"go.mongodb.org/mongo-driver/mongo"
+	"strconv"
+
+	"reflect"
+	"runtime"
+	"strings"
+)
+
+// currentFuncName returns the name of the caller function as a string
+// for func Foo() { x := currentFuncName }, x == "Foo"
+func currentFuncName() string {
+	current := make([]uintptr, 1)
+	if level := runtime.Callers(2, current); level == 0 {
+		return ""
+	}
+	caller := runtime.FuncForPC(current[0] - 1)
+	if caller == nil {
+		return ""
+	}
+	sysFuncName := caller.Name()
+	return partitionByDelimiterAtIndex(sysFuncName, ".", -1)
+
+}
+
+func startMongoEvent(opName string, coll *MongoCollectionWrapper) *protocol.Event {
+	defer epsagon.GeneralEpsagonRecover("mongo-driver", currentFuncName(), coll.tracer)
+	return &protocol.Event{
+		Id:        "mongodb-" + uuid.New().String(),
+		Origin:    "mongodb",
+		ErrorCode: protocol.ErrorCode_OK,
+		StartTime: tracer.GetTimestamp(),
+		Resource:  createMongoResource(opName, coll),
+	}
+}
+
+func completeMongoEvent(currentTracer tracer.Tracer, event *protocol.Event) {
+	defer epsagon.GeneralEpsagonRecover("mongo-driver", currentFuncName(), currentTracer)
+	event.Duration = tracer.GetTimestamp() - event.StartTime
+	currentTracer.AddEvent(event)
+
+}
+
+func createMongoResource(opName string, coll *MongoCollectionWrapper) *protocol.Resource {
+	defer epsagon.GeneralEpsagonRecover("mongo-driver", currentFuncName(), coll.tracer)
```

Comment:
I know it was my comment, but actually we shouldn't use it inside the internal funcs, as we won't catch panics from the user original code.
Let's talk about it  :)

Reply:
let's put the function name hard coded, I think its better than running this code at each call to the mongo wrapper